.github/workflows/DocsNav.yml

```
@@ -1,49 +1,37 @@
-name: Add Navbar
+name: Rebuild docs with newest navbar
 
 on:
-  page_build:  # Triggers the workflow on push events to gh-pages branch
-  workflow_dispatch:  # Allows manual triggering
+  # 3:25 AM UTC every Sunday -- choose an uncommon time to avoid
+  # periods of heavy GitHub Actions usage
   schedule:
-    - cron: '0 0 * * 0'  # Runs every week on Sunday at midnight (UTC)
+    - cron: '25 3 * * 0'
+  # Whenever needed
+  workflow_dispatch:
+
+permissions:
+  contents: write
 
 jobs:
-  add-navbar:
+  update-navbar:
     runs-on: ubuntu-latest
-    permissions:
-      contents: write
+
     steps:
-      - name: Checkout gh-pages
+      - name: Checkout gh-pages branch
         uses: actions/checkout@v4
         with:
           ref: gh-pages
-          fetch-depth: 0
 
-      - name: Download insert_navbar.sh
-        run: |
-          curl -O https://raw.githubusercontent.com/TuringLang/turinglang.github.io/main/assets/scripts/insert_navbar.sh
-          chmod +x insert_navbar.sh
+      - name: Insert navbar
+        uses: TuringLang/actions/DocsNav@main
+        with:
+          doc-path: '.'
 
-      - name: Update Navbar
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Commit and push changes
         run: |
-          git config user.name github-actions[bot]
-          git config user.email github-actions[bot]@users.noreply.github.com
-
-          # Define the URL of the navbar to be used
-          NAVBAR_URL="https://raw.githubusercontent.com/TuringLang/turinglang.github.io/main/assets/scripts/TuringNavbar.html"
-
-          # Update all HTML files in the current directory (gh-pages root)
-          ./insert_navbar.sh . $NAVBAR_URL
-
-          # Remove the insert_navbar.sh file
-          rm insert_navbar.sh
-
-          # Check if there are any changes
           if [[ -n $(git status -s) ]]; then
-            git add .
-            git commit -m "Added navbar and removed insert_navbar.sh"
-            git push "https://${GITHUB_ACTOR}:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git" gh-pages
+            git add -A
```

```suggestion
            git config user.name github-actions[bot]
            git config user.email github-actions[bot]@users.noreply.github.com
            git add -A
```

git push still needs a name + email, I ran into this error earlier. I guess the version with the GITHUB_ACTOR should work too actually?

---

I just got this error too😅: https://github.com/TuringLang/AdvancedMH.jl/actions/runs/13224890742/job/36914500469